machineconfiguration/v1alpha1/types_machineconfignode.go

```diff
@@ -0,0 +1,157 @@
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:Pattern=`^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])(\.([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9]))*$`
+	// +kubebuilder:validation:Required
+	Desired string `json:"desired"`
+}
+
+// MachineConfigNodeSpecMachineConfigVersion holds the desired config version for the current observed machine config node.
+// When Current != Desired; the MachineConfigOperator is in an upgrade phase and the machine config node will
+// take account of upgrade related events. Otherwise they will be ignored given that certain operations
+// happen both during the MCO's upgrade mode and the daily operations mode.
+type MachineConfigNodeSpecMachineConfigVersion struct {
+	// desired is the name of the machine config that the the node should be upgraded to.
+	// This value is set when the machine config pool generates a new version of its rendered configuration.
+	// When this value is changed, the machine config daemon starts the node upgrade process.
+	// If the transition of this value is accepted, the new desired version will be updated in the .status.configVersion.desired field
+	// and the upgrade will proceed.
+	// Must be a lowercase RFC-1123 hostname (https://tools.ietf.org/html/rfc1123)
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:Pattern=`^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])(\.([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9]))*$`
+	// +kubebuilder:validation:Required
+	Desired string `json:"desired"`
+}
+
+// StateProgress is each possible state for each possible MachineConfigNodeType
+// UpgradeProgression Kind will only use the "MachinConfigPoolUpdate..." types for example
+type StateProgress string
+
+const (
+	// MachineConfigPoolUpdatePrepared describes a machine that is preparing in the daemon to trigger an update
+	MachineConfigPoolUpdatePrepared StateProgress = "UpdatePrepared"
+	// MachineConfigPoolUpdateInProgress describes a machine that is in progress of updating
+	MachineConfigPoolUpdateInProgress StateProgress = "UpdateInProgress"
+	// MachineConfigPoolUpdatePostAction describes a machine that is executing its post update action
+	MachineConfigPoolUpdatePostAction StateProgress = "UpdatePostAction"
+	// MachineConfigPoolUpdateCompleted describes a machine that is in the process of resuming normal processes
+	MachineConfigPoolUpdateCompleted StateProgress = "UpdateCompleted"
+	// MachineConfigPoolUpdateComplete describes a machine that has a matching desired and current config after executing an update
+	MachineConfigPoolUpdateComplete StateProgress = "Updated"
+	// MachineConfigPoolUpdateResumed describes a machine that is in the process of resuming normal processes
+	MachineConfigPoolResumed StateProgress = "Resumed"
+	// MachineConfigNodeErrored describes when a machine had run into an issue
+	MachineConfigNodeErrored StateProgress = "Errored"
```

Comment:
Are we still keeping this one, i thought we concluded we wouldn't need this?

Reply:
removed this